mypy_extensions.py

```diff
@@ -36,8 +36,15 @@ def _typeddict_new(cls, _typename, _fields=None, **kwargs):
     elif kwargs:
         raise TypeError("TypedDict takes either a dict or keyword arguments,"
                         " but not both")
-    return _TypedDictMeta(_typename, (), {'__annotations__': dict(_fields),
-                                          '__total__': total})
+
+    ns = {'__annotations__': dict(_fields), '__total__': total}
+    try:
+        # Setting correct module is necessary to make typed dict classes pickleable.
+        ns['__module__'] = sys._getframe(1).f_globals.get('__name__', '__main__')
```

Comment:
I trust that you tested this, but could you please add a test for this? Probably just check `.__module__` in `test_py36_class_syntax_usage()` (rebase on master to get the recently added tests).

Reply:
I added the test. And there are already existing tests that validate `__module__` when using instantiation syntax.